### PR TITLE
Update to Retrofit 2.0.0-beta3, OkHttp 3.0.0-RC1.

### PR DESCRIPTION
**Pull request by JakeWharton**

### CODE REVIEW

File: src/internalDebug/java/com/jakewharton/u2020/ui/debug/DebugView.java [hunk 1]
```diff
@@ -249,11 +248,14 @@ private void setupNetworkSection() {
...
-            networkProxyAddress.delete();
-            client.setProxy(null);
-            apiClient.setProxy(null);
+            // Only clear the proxy and restart if one was previously set.
+            if (currentProxyPosition != ProxyAdapter.NONE) {
+              Timber.d("Clearing network proxy");
+              // TODO: Keep the custom proxy around so you can easily switch back and forth.
+              networkProxyAddress.delete();
+              // Force a restart to re-initialize the app without a proxy.
+              ProcessPhoenix.triggerRebirth(getContext());
```
Comment (mattprecious) [inline src/internalDebug/java/com/jakewharton/u2020/ui/debug/DebugView.java]: Not swappable on the fly anymore?

**Comment by mattprecious** [comment]
lgtm!

File: src/internalDebug/java/com/jakewharton/u2020/ui/debug/DebugView.java (hunk 1 above)
Comment (JakeWharton) [inline src/internalDebug/java/com/jakewharton/u2020/ui/debug/DebugView.java]: No. `OkHttpClient` is immutable and I don't want to pipe around a `Provider<OkHttpClient>` everywhere.

File: build.gradle [hunk 2]
```diff
@@ -132,15 +132,14 @@ dependencies {
...
+  compile 'com.squareup.okhttp3:okhttp:3.0.0-RC1'
+  compile 'com.squareup.okhttp3:logging-interceptor:3.0.0-RC1'
   compile 'com.squareup.picasso:picasso:2.5.2'
-
-  compile 'com.squareup.retrofit:retrofit:2.0.0-beta2'
-  compile 'com.squareup.retrofit:converter-moshi:2.0.0-beta2'
-  compile 'com.squareup.retrofit:adapter-rxjava:2.0.0-beta2'
-  internalDebugCompile 'com.squareup.retrofit:retrofit-mock:2.0.0-beta2'
-  internalDebugCompile 'com.squareup.retrofit:adapter-rxjava-mock:2.0.0-beta2'
+  compile 'com.jakewharton.picasso:picasso2-okhttp3-downloader:1.0.1'
```
Comment (tasomaniac) [inline build.gradle]: Should OkHttp 2.0 be disabled from Picasso's dependencies in this case?

Comment (JakeWharton) [inline build.gradle]: It's an optional dependency on Picasso so it won't bring it in transitively.

Comment (tasomaniac) [inline build.gradle]: That's what I thought first but then I saw I can access OkHttp 2. Maybe I should clean the project. Thank you.

**Pull request merged**